File: JavaProject/ZJU_Java/clock/src/display/led/Display.java
package display.led;

public class Display {
	private int value = 0;
	private int limit = 0;
	private static int step = 1;

	public  Display(int limit) {
		this.limit = limit;
	}

	public void increase() {
		value++;
		if (value == limit) {
			value = 0;

		}
	}

	public int getValue() {
		return value;
	}

	

	public static void main(String[] args) {
		
	}
}

File: c++/oop/firstclass/AutoClock/Display.cpp
#include <iostream>
using namespace std;

class Display {
private:
    int value=0,limit=0;
public:
    Display(int limit) {
        this->limit = limit;
    }
    void increase() {
        value += 1;
        if (value == limit) {
            value = 0;
        }
    }

    int getValue() {
        return value;
    }

    Display(Display &display) {
       /* value = display.value;
        limit = display.limit;*/
        cout << "Display对象被复制了" << endl;
    }
};


